tests/integration_tests/test_observation_times.py

```diff
@@ -0,0 +1,159 @@
+"""
+Tests behavior of matching response times to observation times
+"""
+from contextlib import redirect_stderr
+from datetime import datetime
+from io import StringIO
+from textwrap import dedent
+
+import hypothesis.strategies as st
+import numpy as np
+import pytest
+from hypothesis import assume, given, settings
+
+from ert.cli import ES_MDA_MODE
+from ert.cli.main import ErtCliError
+from tests.unit_tests.config.observations_generator import summary_observations
+from tests.unit_tests.config.summary_generator import summaries
+
+from .run_cli import run_cli
+
+start = datetime(1969, 1, 1)
+observation_time = datetime(2024, 1, 1)
+epsilon = 0.1
+
+
+@settings(max_examples=3)
+@given(
+    responses=st.lists(
+        summaries(
+            start_date=st.just(start),
+            time_deltas=st.just(
+                [((observation_time - start).total_seconds() + epsilon) / 3600]
+            ),
+            summary_keys=st.just(["FOPR"]),
+            use_days=st.just(False),
+        ),
+        min_size=2,
+        max_size=2,
+    ),
+    observation=summary_observations(
+        summary_keys=st.just("FOPR"),
+        std_cutoff=10.0,
+        names=st.just("FOPR_OBSERVATION"),
+        dates=st.just(observation_time),
+        time_types=st.just("date"),
+    ),
+    std_cutoff=st.floats(min_value=0.0, max_value=1.0),
+    enkf_alpha=st.floats(min_value=3.0, max_value=10.0),
+)
+def test_small_time_mismatches_are_ignored(
+    responses, observation, tmp_path_factory, std_cutoff, enkf_alpha
+):
+    tmp_path = tmp_path_factory.mktemp("summary")
+    (tmp_path / "config.ert").write_text(
+        dedent(
+            f"""
+            NUM_REALIZATIONS 2
+            ECLBASE CASE
+            SUMMARY FOPR
+            MAX_SUBMIT 1
+            GEN_KW KW_NAME prior.txt
+            OBS_CONFIG observations.txt
+            STD_CUTOFF {std_cutoff}
+            ENKF_ALPHA {enkf_alpha}
+            """
+        )
+    )
+    (tmp_path / "prior.txt").write_text("KW_NAME NORMAL 0 1")
+    response_values = np.array(
+        [r[1].steps[-1].ministeps[-1].params[-1] for r in responses]
+    )
+    std_dev = response_values.std(ddof=0)
+    assume(np.isfinite(std_dev))
+    assume(std_dev > std_cutoff)
+    observation.value = float(response_values.mean())
+    for i in range(2):
+        for j in range(4):
+            summary = responses[i]
+            smspec, unsmry = summary
+            (tmp_path / f"simulations/realization-{i}/iter-{j}").mkdir(parents=True)
+            smspec.to_file(
+                tmp_path / f"simulations/realization-{i}/iter-{j}/CASE.SMSPEC"
+            )
+            unsmry.to_file(
+                tmp_path / f"simulations/realization-{i}/iter-{j}/CASE.UNSMRY"
+            )
+    (tmp_path / "observations.txt").write_text(str(observation))
+
+    stderr = StringIO()
+    with redirect_stderr(stderr):
+        run_cli(ES_MDA_MODE, str(tmp_path / "config.ert"), "--weights=0,1")
```

Comment:
If you want to do this faster, it is possible to mock this by doing `load_from_forward_model` combined with a manual update.

Reply:
Not too many calls no, but adds complexity yes, not sure it is worth the trade-off, but because it is a property test it might be.